core/src/main/scala/org/apache/spark/storage/BlockObjectWriter.scala

```diff
@@ -189,13 +189,12 @@ private[spark] class DiskBlockObjectWriter(
         objOut.flush()
         bs.flush()
         close()
-      }
-
-      val truncateStream = new FileOutputStream(file, true)
-      try {
-        truncateStream.getChannel.truncate(initialPosition)
-      } finally {
-        truncateStream.close()
+        val truncateStream = new FileOutputStream(file, true)
```

Comment:
I moved this into the `if (initialized)` block because there should be no need to truncate unless we've written to the file.